logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java

```diff
@@ -432,30 +440,29 @@ private void doCreateSnapshot() {
   }
 
   private void writeSnapshot(final long eventPosition) {
-    SnapshotWriter snapshotWriter = null;
+    final SnapshotMetadata metadata =
+        new SnapshotMetadata(
+            eventPosition,
+            lastWrittenEventPosition,
+            streamProcessorContext.getLogStream().getTerm(),
+            false);
+
     try {
       final long start = System.currentTimeMillis();
       final String name = streamProcessorContext.getName();
       LOG.info("Write snapshot for stream processor {} at event position {}.", name, eventPosition);
 
-      snapshotWriter = snapshotStorage.createSnapshot(name, eventPosition);
-
-      final long snapshotSize = snapshotWriter.writeSnapshot(streamProcessor.getStateResource());
-      snapshotWriter.commit();
+      snapshotController.takeSnapshot(metadata);
 
       final long snapshotCreationTime = System.currentTimeMillis() - start;
       LOG.info("Creation of snapshot {} took {} ms.", name, snapshotCreationTime);
 
-      metrics.recordSnapshotSize(snapshotSize);
+      // metrics.recordSnapshotSize(snapshotSize);
```

Comment:
It's a little more complicated to get the size of a snapshot now. Snapshots (or checkpoints as RocksDB call them - snapshots in RocksDB are something else) are essentially soft copies of your running database.

A RocksDB database is essentially: in memory buffers (memtable or write buffers, bloom filter, indexes, caches, etc.), immutable on-disk SST files, and a bunch of metadata files (and possibly a WAL). When you take a checkpoint, RocksDB will create hardlink of the immutable files in a new folder, will flush the in memory contents to the new folder, update metadata there based on that, and potentially apply the write-ahead-log (WAL) for it. So most of the data between your checkpoint and the running DB is shared, with the overhead being the size of the flushed data, essentially (and a few other negligible things as far as I can tell). 

So the real size of your snapshot is the amount of data that could potentially be discarded, I guess? In a running system, most of it will be shared across many snapshots + the current running database. So what do we want to report here? This is why I have it commented out.

Speaking of, there's a lot of metrics we can extract which could be interesting, listed as properties in RocksDB: https://github.com/facebook/rocksdb/blob/54de56844d0a0bd6555fb0851a52ab0752987370/include/rocksdb/db.h#L429

But I don't think it's necessary for now. I will remove the commented out line though, but we can think later what kind of metrics we want to report.

Reply:
I agree. Please create an issue for this topic.